private/bufpkg/bufmodule/module_file_set.go

```diff
@@ -57,10 +57,27 @@ func newModuleFileSet(
 			),
 		)
 	}
+	multiReadBucket := newMultiModuleReadBucket(moduleReadBuckets...)
+	seenPaths := map[string][]string{}
+	if err := multiReadBucket.WalkModuleFiles(context.Background(), "", func(moduleObjectInfo *moduleObjectInfo) error {
+		modulePrefixedPaths, ok := seenPaths[moduleObjectInfo.Path()]
+		modulePrefixedPath := moduleObjectInfo.ExternalPath()
+		if moduleObjectInfo.ModuleIdentity() != nil {
+			modulePrefixedPath = moduleObjectInfo.ModuleIdentity().IdentityString() + "#" + modulePrefixedPath
+		}
+		modulePrefixedPaths = append(modulePrefixedPaths, modulePrefixedPath)
+		if ok {
+			return storage.NewErrExistsMultipleLocations(moduleObjectInfo.Path(), modulePrefixedPaths...)
+		}
+		seenPaths[moduleObjectInfo.Path()] = modulePrefixedPaths
+		return nil
```

Comment:
Whenever possible, it's best to check the `ok` immediately after it's created. We can reorganize this a bit in its current state, but how we do might depend on the comment above.

Reply:
Sounds good, I'll keep this in mind as I restructure this based on the comment above :)